### PR TITLE
Enhance e2e tests to check backend for traces

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
@kevinearls, @objectiser , could one of you please review this?

File: test/check-trace.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    # we expect at least one trace and at least two spans in the first trace
+    if (( numTraces > 0 )); then
+        echo "found at least one trace for the e2e-test service" >> ./test/test.log
+        if (( numSpans > 1 )); then
+            echo "found at least two spans in the trace" >> ./test/test.log
+            echo "✅ Traces found in the backend."
+            break
+        else
+            echo "not enough spans in the trace" >> ./test/test.log
```
Comment (kevinearls) [inline test/check-trace.sh:19]: nit, and a completely optional one.  It *might* be useful to print the actual number of spans received.  Information like that is often useful in debugging test failures.

**Review by kevinearls — APPROVED** [review]
LGTM

File: test/check-trace.sh (hunk 1 above)
Comment (jpkrohling) [reply test/check-trace.sh]: Done!

**Pull request merged**